src/options.rs

```diff
@@ -1,71 +1,181 @@
-use gumdrop::Options;
+use anyhow::{Error, Result};
+use serde::Deserialize;
+use std::{fs, io::ErrorKind};
+use structopt::StructOpt;
+
+// Macro for generating default functions to be used by serde
+macro_rules! default_function {
+    ( $( $name:ident : $T:ty = $e:expr; )* ) => {
+        $(
+            fn $name() -> $T {
+                $e
+            }
+        )*
+    };
+}
+
+// Macro for merging configuration values
+macro_rules! fold_in {
```

Comment:
No, its just for parsing. The flattening I was referring to means that you can define two structs and have them be represented as flat command line arguments instead of requiring subcommands. This is useful since it means I don't need to define stuff multiple times.

Reply:
Gotcha. Thanks for the clarification.